### ISSUE TITLE
Kokkos unit test failing with MPICH

### ISSUE THREAD

**Issue by prwolfe** [trilinos/Trilinos#2793]
When building with MPICH3.2 I see a failure in KokkosCore_UnitTest_Serial.exe (the serial.cxx test) due to comparison of two doubles.  It looks like they are off by less than 10-9 and I'm not sure what if any tolerance is acceptable in this case.

@trilinos/kokkos 

## Expectations
This test should pass with all mpi variants

## Current Behavior
This test passes in openmpi builds

## Steps to Reproduce
To recreate this you can use the PR template we are working on. I did attempt to recreate this In a kokkos stand-alone build without sucess.

```
source ${TrilinosSource}/cmake/std/sems/PullRequestIntel17.0.1TestingEnv.sh
cmake -C ${TrilinosSource}/cmake/std/PullRequestLinuxIntelTestingSettings.cmake -DTrilinos_ENABLE_Kokkos=ON ${TrilinosSource}
cd pacakges/kokkos/core/unit_tests
make
```

**Comment by ibaned** [comment]
@prwolfe since this is assumed to be a Kokkos issue, I moved it to the Kokkos repository: kokkos/kokkos#1632

**Comment by prwolfe** [comment]
The fix for this is now in Trilinos (Since November 5th looks like) so we need to re-enable the test and close this.

**Comment by prwolfe** [comment]
resolved by #5375
